src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -644,7 +644,7 @@ private async Task CreateExecutablesAsync(IEnumerable<AppResource> executableRes
                     }
                 }
 
-                spec.Env = new();
+                spec.Env = new(config.Count);
```

Comment:
I didn't mean to commit this unrelated change, but if we're happy with it then I'll leave it. It just sets a list capacity to avoid redundant allocations.

Reply:
Ended up backing this out.